src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -11,6 +11,15 @@
 #include <java_bytecode/java_types.h>
 #include <java_bytecode/java_utils.h>
 
+static std::string java_type_simplify(const std::string &fqn_java_type)
+{
+  const std::string java_lang("java::java.lang");
+  const std::string package_name(java_class_to_package(fqn_java_type));
+  if(package_name==java_lang)
+    return fqn_java_type.substr(java_lang.length()+1);
```

Comment:
Rename "pretty-print" or something, since this produces types that should *only* be used for human readable output. Perhaps move this into type2cleanjava?